core/src/test/scala/unit/kafka/network/RequestChannelTest.scala

```diff
@@ -362,11 +362,13 @@ class RequestChannelTest {
     val props = new Properties()
     props.put(KafkaConfig.ZkConnectProp, "127.0.0.1:2181")
     props.put(KafkaConfig.RequestMetricsTotalTimeBucketsProp, "0, 10, 30, 300")
+    props.put(KafkaConfig.TotalTimeHistogramEnabledMetricsProp, util.Arrays.asList("Produce0To1MbAcks1", "Produce0To1MbAcksAll", "FetchConsumer0To1Mb"))
+
     val config = KafkaConfig.fromProps(props)
-    // totalTimeBucketHist is only enabled for Produce/Fetch related requests.
+    // totalTimeBucketHist is only enabled for RequestMetrics with name defined in TotalTimeHistogramEnabledMetricsProp.
     val requestMetricsNoTotalTimeBucketHist = new RequestMetrics("RequestMetrics", config)
     assertTrue(requestMetricsNoTotalTimeBucketHist.totalTimeBucketHist.isEmpty)
-    val requestMetrics = new RequestMetrics("FetchConsumer", config)
+    val requestMetrics = new RequestMetrics("FetchConsumer0To1Mb", config)
     assertTrue(requestMetrics.totalTimeBucketHist.isDefined)
```

Comment:
Can you add a corresponding negative check (if one doesn't already exist, didn't spot any further down) that metrics NOT listed in the property don't exist?  (I don't think that's what the isEmpty assertion above does, right?)

Reply:
The isEmpty assertion is the negative test. If the requestMetrics has the name (in this negative case the name is "RequestMetrics") that is not in the config, there would be no totalTimeBucketHist.